examples/restrictable-expression.flix

```diff
@@ -0,0 +1,62 @@
+restrictable enum Expr[s] {
+    case Cst(Bool)
+    case Var(Int32)
+    case Not(Expr[s])
+    case And(Expr[s], Expr[s])
+    case Or(Expr[s], Expr[s])
+    case Xor(Expr[s], Expr[s])
+}
+
+def size(e: Expr[s]): Int32 = choose e {
+    case Expr.Cst(_) => 1
+    case Expr.Var(_) => 1
+    case Expr.Not(x) => size(x) + 1
+    case Expr.And(x, y) => size(x) + size(y) + 1
+    case Expr.Or(x, y) => size(x) + size(y) + 1
+    case Expr.Xor(x, y) => size(x) + size(y) + 1
+}
+
+def eq(e1: Expr[s1], e2: Expr[s2]): Bool = ??? // nested choose
```

Comment:
What this for?

Reply:
Just a cool function i wanted to write, I can remove it/ write it